File: scala_project/Point.scala
// 定义一个表示二维坐标点的Scala样例类
case class Point(x: Double, y: Double)

// 伴生对象，包含一些辅助方法
object Point {
  // 创建一个原点 (0, 0)
  def origin: Point = Point(0.0, 0.0)

  // 计算两个点之间的欧几里得距离
  def distance(p1: Point, p2: Point): Double = {
    math.sqrt(math.pow(p2.x - p1.x, 2) + math.pow(p2.y - p1.y, 2))
  }
}

File: Point.swift
import Foundation

struct Point {
    var x: Double
    var y: Double
    
    init(x: Double, y: Double) {
        self.x = x
        self.y = y
    }
    
    // Optional: A method to calculate distance from origin
    func distanceFromOrigin() -> Double {
        return sqrt(x * x + y * y)
    }
    
    // Optional: A method to calculate distance to another point
    func distance(to other: Point) -> Double {
        let dx = self.x - other.x
        let dy = self.y - other.y
        return sqrt(dx * dx + dy * dy)
    }
}